include/spdlog/sinks/eventlog_sink_win32.h

```diff
@@ -0,0 +1,246 @@
+// Copyright(c) 2020 bandana2004@gmail.com
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+
+namespace eventlog_sink_win32 {
+
+struct win32_error : public spdlog_ex
+{
+    static std::string format(std::string const& func, DWORD error = GetLastError())
+    {
+        return fmt::format(func + " failed ({0})", error);
+    }
+
+    static void report(char const* message) SPDLOG_NOEXCEPT
+    {
+        fprintf(stderr, "%s", message);
+        fflush(stderr);
+    }
+
+    static void report(std::string const& message) SPDLOG_NOEXCEPT
+    {
+        report(message.c_str());
+    }
+
+    win32_error(std::string const& func_name, DWORD error = GetLastError())
+        : spdlog_ex(format(func_name, error), error)
+    {}
+};
+
+struct sid
+{
+    static SID* duplicate(SID* p_src_sid)
+    {
+        if (!IsValidSid(p_src_sid))
+            SPDLOG_THROW(spdlog_ex("sid::duplicate: invalid SID received"));
+
+        DWORD sid_size = ::GetLengthSid(p_src_sid);
+        SID* p_dest_sid = (SID*) ::HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sid_size);
+
+        if (!p_dest_sid)
+            SPDLOG_THROW(win32_error("HeapAlloc"));
+
+        if (!CopySid(sid_size, p_dest_sid, p_src_sid))
+        {
+            sid::free(&p_dest_sid);
+            SPDLOG_THROW(win32_error("CopySid"));
+        }
+
+        return p_dest_sid;
+    }
+
+    static void free(SID** ppsid)
+    {
+        if (ppsid && *ppsid)
+        {
+            if (!HeapFree(GetProcessHeap(), 0, (LPVOID) *ppsid))
+            {
+                SPDLOG_THROW(win32_error("HeapFree"));
+            }
+        }
+
+        *ppsid = nullptr;
+    }
+
+    static SID* get_current_user_sid()
+    {
+        struct process_token_t
```

Comment:
Please add comments about all those window functions - its very hard to understand the purpose